hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/TableSnapshotInputFormatImpl.java

```diff
@@ -392,17 +405,41 @@ public static Scan extractScanFromConf(Configuration conf) throws IOException {
       SNAPSHOT_INPUTFORMAT_SCAN_METRICS_ENABLED_DEFAULT);
     scan.setScanMetricsEnabled(scanMetricsEnabled);
 
+    boolean useRegionLoc = conf.getBoolean(SNAPSHOT_INPUTFORMAT_LOCALITY_BY_REGION_LOCATION,
+      SNAPSHOT_INPUTFORMAT_LOCALITY_BY_REGION_LOCATION_DEFAULT);
+
+    Connection connection;
+    RegionLocator regionLocator = null;
+    if (localityEnabled && useRegionLoc) {
+      connection = ConnectionFactory.createConnection(new Configuration(conf));
```

Comment:
Closing this connection?
Also on this connection, we will make calls to read from hbase:meta only. We can set connection executor pool size with lowering value for 'hbase.hconnection.threads.max'?  We allow core threads to die but initially will create more threads (256).

Reply:
Good idea, let me upload a new patch.

Reply:
Hi @anoopsjohn, I pushed updates based on your comment, please take a look, thanks.